torch/csrc/cuda/comm.cpp

```diff
@@ -34,38 +35,75 @@ struct unique_type_checker {
   bool unique = true;
 };
 
-std::vector<Tensor> broadcast(const Tensor& tensor, IntArrayRef devices) {
-  if (tensor.is_cuda() && tensor.get_device() != devices[0])
-    throw std::runtime_error("device of broadcasted tensor must appear as the "
-                             "first on devices list");
-  std::vector<Tensor> tensors;
-  tensors.reserve(devices.size());
+// ***************** Broadcast *******************
+//
+// Broadcast a source tensor (CPU or CUDA) to a list of CUDA devices, or CUDA
+// tensors on one or more devices.
+
+// no checks
+static inline
+std::vector<Tensor>& _broadcast_out_impl(const Tensor& tensor, std::vector<Tensor> &out_tensors) {
```

Comment:
curious, since the `out_tensors` is already in the arg, why do we need to return it again?

Reply:
We don't need to! This can have a `void` return type. I just followed the python `out=` and inplace functions signatures and I don't think it matters.